Assets/UIElements/_Elements/XRUI_Buttons/Scripts/XRUI_2DButtonSprite.cs

```diff
@@ -44,25 +45,39 @@ protected override void Awake()
         base.Awake();
         SetSpriteColors();
 
-
         xrFeedback.onTouchEnter.AddListener((XRController controller) => { onClickDown?.Invoke(); });
         xrFeedback.onTouchExit.AddListener((XRController controller) => { onClickUp?.Invoke(); });
     }
+    private void Start()
+    {
+        if (xrFeedback.allowDistanceEvents)
+        {
+            XRBaseInteractable interactable = gameObject.GetComponent<XRBaseInteractable>();
+            if (interactable == null)
+                interactable = gameObject.GetComponentInChildren<XRBaseInteractable>();
 
+            if (interactable != null)
+                interactable.onSelectEnter.AddListener((XRBaseInteractor) => { onClickDown?.Invoke(); });
+        }
+        else if (distanceCollider != null)
+        {
+            distanceCollider.SetActive(false);
+        }
+    }
```

Comment:
Maybe we can do this on the XRUI_Base class, or create a abstract function to implement specifically the Distance Events

Reply:
Changed